internal/operator/controller/gatewayclass/controller.go

```diff
@@ -56,12 +57,31 @@ func New(mgr manager.Manager) (controller.Controller, error) {
 	if err != nil {
 		return nil, err
 	}
-	if err := c.Watch(&source.Kind{Type: &gatewayv1alpha1.GatewayClass{}}, &handler.EnqueueRequestForObject{}); err != nil {
+	// Only enqueue GatewayClass objects that specify the operator as the controller.
+	if err := c.Watch(&source.Kind{Type: &gatewayv1alpha1.GatewayClass{}}, r.enqueueRequestForGatewayClass()); err != nil {
 		return nil, err
 	}
 	return c, nil
 }
 
+// enqueueRequestForGatewayClass returns an event handler that maps events to
+// GatewayClass objects owned by the operator.
+func (r *reconciler) enqueueRequestForGatewayClass() handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(a client.Object) []reconcile.Request {
+		gc := a.(*gatewayv1alpha1.GatewayClass)
```

Comment:
nit: To be defensive, check the type assertion succeeded here

```go
gc, ok := a.(*gatewayv1alpha1.GatewayClass)
if ok && objgc.IsController(gc) {
...
}
return []reconcile.Request{}
```

Practically it should never happen, but just to be safe